tests/ui/traits/on_unimplemented_long_types.stderr

```diff
@@ -13,8 +13,6 @@ LL | |         ))))))))))),
 LL | |     )))))))))))
    | |_______________- return type was inferred to be `Option<Option<Option<...>>>` here
    |
-   = note: the full name for the type has been written to '$TEST_BUILD_DIR/traits/on_unimplemented_long_types/on_unimplemented_long_types.long-type-hash.txt'
-   = note: consider using `--verbose` to print the full type name to the console
```

Comment:
Another benefit of making `Diag` carry the path: it's impossible to have redundant notes now.